org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/model/ModelPersistAction.java

```diff
@@ -0,0 +1,152 @@
+/*-
+ *******************************************************************************
+ * Copyright (c) 2011, 2016 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.scanning.device.ui.model;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.richbeans.widgets.file.FileSelectionDialog;
+import org.eclipse.scanning.api.IModelProvider;
+import org.eclipse.scanning.api.stashing.IStashing;
+import org.eclipse.scanning.device.ui.Activator;
+import org.eclipse.scanning.device.ui.ServiceHolder;
+import org.eclipse.swt.widgets.Display;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * This action can be used to allow scan paths to 
+ * be saved and loaded.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public class ModelPersistAction<T> extends Action {
+	
+	private static final Logger logger = LoggerFactory.getLogger(ModelPersistAction.class);
+
+	public static final String[] IDS = new String[]{"SAVE", "LOAD"};

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this member "protected". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2386)